src/exla/exla_nif_util.cc

```diff
@@ -74,6 +74,39 @@ namespace exla {
     return 1;
   }
 
+  // TODO: Accept list, not tuple
+  int get_argument_layouts(ErlNifEnv* env, ERL_NIF_TERM tuple, absl::Span<xla::Shape*> &span){
+    const ERL_NIF_TERM* elems;
+    int num_elems;
+    if(!enif_get_tuple(env, tuple, &num_elems, &elems)) return 0;
+    xla::Shape* data[num_elems];
+    for(int i=0;i<num_elems;i++){
+      xla::Shape* elem;
+      if(!get<xla::Shape>(env, elems[i], elem)) return 0;
+      data[i] = elem;
+    }
+    span = absl::Span<xla::Shape*>(data, num_elems);
+    return 1;
+  }
+
+  int get_arguments(ErlNifEnv* env, ERL_NIF_TERM tuple, absl::Span<xla::ShapedBuffer*> &span){
+    const ERL_NIF_TERM* elems;
+    int num_elems;
+    if(!enif_get_tuple(env, tuple, &num_elems, &elems)) return 0;
+    xla::ShapedBuffer* data[num_elems];
+    for(int i=0;i<num_elems;i++){
+      xla::ShapedBuffer* elem;
+      if(!get<xla::ShapedBuffer>(env, elems[i], elem)) return 0;
+      data[i] = elem;
+    }
+    span = absl::Span<xla::ShapedBuffer*>(data, num_elems);
+    // TODO: For some reason, this prevents `run` from SegFaulting...
+    for(auto i=span.begin();i!=span.end();++i){
+      LOG(ERROR) << (**i);
```

Comment:
This is another one of those SegFaults that happens under random conditions. We don't ever SegFault with 0 arguments. Without this for loop, we SegFault with 1 argument. With this for loop, we don't SegFault with 1 argument, but we SegFault with 2 arguments.

Reply:
Since we suspect a lot of the segfaults are related to the GC, you can use `:erlang.garbage_collect(self())` before and after some of these operations to see if it makes it more consistent.

Reply:
As it turns out, it was due to how I was using the `absl::Span` to handle argument inputs. I switched everything up to use an `std::vector` instead and it worked.